device/coordinate_manager.cpp

```diff
@@ -6,184 +6,462 @@
+void CoordinateManager::fill_pcie_logical_to_translated() {
+    for (size_t x = 0; x < eth_grid_size.x; x++) {
+        for (size_t y = 0; y < eth_grid_size.y; y++) {
+            const CoreCoord physical_coord = pcie_logical_to_physical[{x, y}];
+            const size_t translated_x = physical_coord.x;
+            const size_t translated_y = physical_coord.y;
+            pcie_logical_to_translated[{x, y}] =
+                CoreCoord(translated_x, translated_y, CoreType::PCIE, CoordSystem::TRANSLATED);
+            pcie_translated_to_logical[{translated_x, translated_y}] =
+                CoreCoord(x, y, CoreType::PCIE, CoordSystem::LOGICAL);
         }
-        virtual_x_to_logical_x[*physical_x_it] = x;
-        physical_x_it++;
     }
 }
 
-#include "blackhole/blackhole_coordinate_manager.h"
-#include "grayskull/grayskull_coordinate_manager.h"
-#include "wormhole/wormhole_coordinate_manager.h"
+#include "umd/device/blackhole_coordinate_manager.h"
+#include "umd/device/blackhole_implementation.h"
+#include "umd/device/grayskull_coordinate_manager.h"
+#include "umd/device/grayskull_implementation.h"
+#include "umd/device/wormhole_coordinate_manager.h"
+#include "umd/device/wormhole_implementation.h"
+
+std::shared_ptr<CoordinateManager> CoordinateManager::get_coordinate_manager(
+    tt::ARCH arch, const size_t tensix_harvesting_mask, const size_t dram_harvesting_mask) {
+    switch (arch) {
+        case tt::ARCH::GRAYSKULL:
+            return get_coordinate_manager(
+                arch,
+                tt::umd::grayskull::TENSIX_GRID_SIZE,
+                tt::umd::grayskull::TENSIX_CORES,
+                tensix_harvesting_mask,
+                tt::umd::grayskull::DRAM_GRID_SIZE,
+                tt::umd::grayskull::DRAM_CORES,
+                dram_harvesting_mask,
+                tt::umd::grayskull::ETH_GRID_SIZE,
+                tt::umd::grayskull::ETH_CORES,
+                tt::umd::grayskull::ARC_GRID_SIZE,
+                tt::umd::grayskull::ARC_CORES,
+                tt::umd::grayskull::PCIE_GRID_SIZE,
+                tt::umd::grayskull::PCIE_CORES);
+        case tt::ARCH::WORMHOLE_B0:
+            return get_coordinate_manager(
+                arch,
+                tt::umd::wormhole::TENSIX_GRID_SIZE,
+                tt::umd::wormhole::TENSIX_CORES,
+                tensix_harvesting_mask,
+                tt::umd::wormhole::DRAM_GRID_SIZE,
+                tt::umd::wormhole::DRAM_CORES,
+                dram_harvesting_mask,
+                tt::umd::wormhole::ETH_GRID_SIZE,
+                tt::umd::wormhole::ETH_CORES,
+                tt::umd::wormhole::ARC_GRID_SIZE,
+                tt::umd::wormhole::ARC_CORES,
+                tt::umd::wormhole::PCIE_GRID_SIZE,
+                tt::umd::wormhole::PCIE_CORES);
+        case tt::ARCH::BLACKHOLE:
+            return get_coordinate_manager(
+                arch,
+                tt::umd::blackhole::TENSIX_GRID_SIZE,
```

Comment:
You seem to pass something from there which is constant and obviously is just need to be known by specific implementation.

Can you:
- Remove the other (with longer signature) get_coordinate_manager
- Make BlackholeCoordinateManager accept only two arguments, harvesting masks
- Make GS and WH managers accept only one argument, tensix harvesting mask
- Move the "#include "umd/device/blackhole_implementation.h"" to blackhole_coordinate_manager.cpp, and then call the base constructor there with these same constants? Same for wh and gs

Reply:
https://github.com/tenstorrent/tt-umd/issues/355